src/hotspot/cpu/x86/c1_MacroAssembler_x86.cpp

```diff
@@ -186,6 +186,14 @@ void C1_MacroAssembler::initialize_header(Register obj, Register klass, Register
 
   if (len->is_valid()) {
     movl(Address(obj, arrayOopDesc::length_offset_in_bytes()), len);
+#ifdef _LP64
+    int base_offset = arrayOopDesc::length_offset_in_bytes() + BytesPerInt;
+    if (!is_aligned(base_offset, BytesPerWord)) {
+      // Clear gap/first 4 bytes following the length field.
+      xorl(t1, t1);
+      movl(Address(obj, arrayOopDesc::length_offset_in_bytes() + BytesPerInt), t1);
+    }
```

Comment:
There's two `arrayOopDesc::length_offset_in_bytes() + BytesPerInt` calculations in this block. Wouldn't it be better to use `base_offset` instead of the second calculation?

Reply:
Right. Fixed.